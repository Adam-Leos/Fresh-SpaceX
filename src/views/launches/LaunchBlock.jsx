import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import './launchBlock.scss';

const LaunchBlock = ({
  flight_number: flightNumber,
  launch_date_unix: unixLaunchDate,
  launch_success: isLaunchSucceeded,
  links,
  rocket,
}) => {
  /* eslint-disable global-require */
  const imageSource = require('./../../img/missing.png');
  /* eslint-enable global-require */

  return (
    <div className="launch-block">
      {isLaunchSucceeded !== null ? <span className="launch-block__status">{isLaunchSucceeded ? 'Succeeded' : 'Failed'}</span> : null}
      <div className="launch-block__info">
        <h3 className="launch-block__heading">Launch #{ flightNumber }</h3>
        <p className="launch-block__description">
          {rocket.rocket_name} in {moment.unix(unixLaunchDate).format('DD MMM, YYYY (H:mma)')}
        </p>
      </div>
      <img src={links.mission_patch || imageSource} alt="launch_patch" className="launch-block__patch" />
    </div>
  );
};

LaunchBlock.propTypes = {
  flight_number: PropTypes.number.isRequired,
  launch_date_unix: PropTypes.number.isRequired,
  launch_success: PropTypes.bool,
  links: PropTypes.shape({
    mission_patch: PropTypes.string,
  }).isRequired,
  rocket: PropTypes.shape({
    rocket_name: PropTypes.string.isRequired,
  }).isRequired,
};

LaunchBlock.defaultProps = {
  launch_success: false,
};

export default LaunchBlock;
