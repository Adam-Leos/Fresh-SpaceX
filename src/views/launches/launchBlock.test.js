import React from 'react';
import LaunchBlock from './LaunchBlock.jsx';

describe('Testing LaunchBlock component', () => {
  const defaultProps = {
    flight_number: 1,
    launch_date_unix: 1523695214905,
    launch_success: true,
    rocket: {
      rocket_name: 'galcon',
    },
    links: {
      mission_patch: '123',
    },
  };

  test('LaunchBlock is matching snapshot', () => {
    const component = shallow(<LaunchBlock {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  test('LaunchBlock properly working with empty links object (no mission patch)', () => {
    const modifiedProps = { ...defaultProps, links: {} };
    const component = shallow(<LaunchBlock {...modifiedProps} />);

    expect(component).toMatchSnapshot();
  });

  test('LaunchBlock properly working with "true", "false" & "null" launch_success', () => {
    const modifiedPropsNull = { ...defaultProps, launch_success: null };
    const modifiedPropsTrue = { ...defaultProps, launch_success: true };
    const modifiedPropsFalse = { ...defaultProps, launch_success: false };
    const componentWithNullSuccess = shallow(<LaunchBlock {...modifiedPropsNull} />);
    const componentWithTrueSuccess = shallow(<LaunchBlock {...modifiedPropsTrue} />);
    const componentWithFalseSuccess = shallow(<LaunchBlock {...modifiedPropsFalse} />);


    expect(componentWithNullSuccess).toMatchSnapshot();
    expect(componentWithTrueSuccess).toMatchSnapshot();
    expect(componentWithFalseSuccess).toMatchSnapshot();
  });
});
