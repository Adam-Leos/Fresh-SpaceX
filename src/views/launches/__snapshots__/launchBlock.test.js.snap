// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LaunchBlock component LaunchBlock is matching snapshot 1`] = `
<div
  className="launch-block"
>
  <span
    className="launch-block__status"
  >
    Succeeded
  </span>
  <div
    className="launch-block__info"
  >
    <h3
      className="launch-block__heading"
    >
      Launch #
      1
    </h3>
    <p
      className="launch-block__description"
    >
      galcon
       in 
      24 Dec, 50253 (8:48am)
    </p>
  </div>
  <img
    alt="launch_patch"
    className="launch-block__patch"
    src="123"
  />
</div>
`;

exports[`Testing LaunchBlock component LaunchBlock properly working with "true", "false" & "null" launch_success 1`] = `
<div
  className="launch-block"
>
  <div
    className="launch-block__info"
  >
    <h3
      className="launch-block__heading"
    >
      Launch #
      1
    </h3>
    <p
      className="launch-block__description"
    >
      galcon
       in 
      24 Dec, 50253 (8:48am)
    </p>
  </div>
  <img
    alt="launch_patch"
    className="launch-block__patch"
    src="123"
  />
</div>
`;

exports[`Testing LaunchBlock component LaunchBlock properly working with "true", "false" & "null" launch_success 2`] = `
<div
  className="launch-block"
>
  <span
    className="launch-block__status"
  >
    Succeeded
  </span>
  <div
    className="launch-block__info"
  >
    <h3
      className="launch-block__heading"
    >
      Launch #
      1
    </h3>
    <p
      className="launch-block__description"
    >
      galcon
       in 
      24 Dec, 50253 (8:48am)
    </p>
  </div>
  <img
    alt="launch_patch"
    className="launch-block__patch"
    src="123"
  />
</div>
`;

exports[`Testing LaunchBlock component LaunchBlock properly working with "true", "false" & "null" launch_success 3`] = `
<div
  className="launch-block"
>
  <span
    className="launch-block__status"
  >
    Failed
  </span>
  <div
    className="launch-block__info"
  >
    <h3
      className="launch-block__heading"
    >
      Launch #
      1
    </h3>
    <p
      className="launch-block__description"
    >
      galcon
       in 
      24 Dec, 50253 (8:48am)
    </p>
  </div>
  <img
    alt="launch_patch"
    className="launch-block__patch"
    src="123"
  />
</div>
`;

exports[`Testing LaunchBlock component LaunchBlock properly working with empty links object (no mission patch) 1`] = `
<div
  className="launch-block"
>
  <span
    className="launch-block__status"
  >
    Succeeded
  </span>
  <div
    className="launch-block__info"
  >
    <h3
      className="launch-block__heading"
    >
      Launch #
      1
    </h3>
    <p
      className="launch-block__description"
    >
      galcon
       in 
      24 Dec, 50253 (8:48am)
    </p>
  </div>
  <img
    alt="launch_patch"
    className="launch-block__patch"
    src={null}
  />
</div>
`;
